fuel-vm/src/checked_transaction.rs

```diff
@@ -204,13 +206,24 @@ pub trait IntoChecked: FormatValidityChecks + Sized {
 }
 
 /// Provides predicate verification functionality for the transaction.
+#[async_trait::async_trait]
 pub trait CheckPredicates: Sized {
     /// Performs predicates verification of the transaction.
     fn check_predicates(
         self,
         params: &ConsensusParameters,
         gas_costs: &GasCosts,
     ) -> Result<Self, CheckError>;
+
+    /// Performs predicates verification of the transaction in parallel.
+    async fn check_predicates_async<
+        E: ParallelExecutor
+            + ParallelExecutor<TaskResult = Result<Word, PredicateVerificationFailed>>,
```

Comment:
If you know the `TaskResult` at this point, it doesn't make a lot of sense to use generic=)